Enable some PyTorch core tests with inductor

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87490

Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add several inductor-specific skips;
3) Enable several core tests for inductor CI;

cc @jansel @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305

desertfire committed:
Enable some PyTorch core tests with inductor

Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add sevearl inductor-specific skips;
3) Enable one of the core test for inductor CI;

[ghstack-poisoned]

desertfire commented:
@pytorchbot rebase

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add sevearl inductor-specific skips;
3) Enable one of the core test for inductor CI;

cc jansel lezcano fdrocha mlazos soumith voznesenskym yanboliang

[ghstack-poisoned]

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add sevearl inductor-specific skips;
3) Enable one of the core test for inductor CI;

cc jansel lezcano fdrocha mlazos soumith voznesenskym yanboliang

[ghstack-poisoned]

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add sevearl inductor-specific skips;
3) Enable one of the core test for inductor CI;

cc jansel lezcano fdrocha mlazos soumith voznesenskym yanboliang

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/opinfo/core.py:
```diff
@@ -1213,6 +1215,11 @@ def get_decorators(self, test_class, test_name, device, dtype):
         return result
 
     def supported_dtypes(self, device_type):
+        if TEST_WITH_TORCHINDUCTOR:
```
This seems like a kludge. Maybe Inductor would be better modeled as two device types, `InductorCPU` and `InductorCUDA`?

desertfire replied:
Dropped this in the new version. Skipping `complex` from CI cmd instead.

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add several inductor-specific skips;
3) Enable several core tests for inductor CI;

[ghstack-poisoned]

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add several inductor-specific skips;
3) Enable several core tests for inductor CI;

[ghstack-poisoned]

voznesenskym commented on test/dynamo/test_repros.py:
```diff
@@ -1015,6 +1015,8 @@ def test_create_rand_mask_from_inputs(self):
         self.assertEqual(cnt.op_count, 8)
 
+    # TODO: make set_rng_state work with fake tensor
+    @patch.object(torch._dynamo.config, "fake_tensor_propagation", False)
```
Xfail it, rather than skip please, so we know when its fixed (unexpected success)

desertfire replied:
I tried to add `@unittest.expectedFailure`, but what makes it complicated is we have `test_no_fake_tensors.py` which patches `torch._dynamo.config. fake_tensor_propagation` to `False` and makes this test unexpectedly succeed. I tried to rewrite the decorator with a conditional checking but that doesn't work well with `object.patch`. Given the unlikelihood this test will unexpectedly get fixed, I will skip adding xfail here.

voznesenskym commented:
Please replace skips w/ xfails

eellison commented on .jenkins/pytorch/test.sh:
```diff
@@ -251,13 +251,8 @@ test_dynamo_shard() {
-  # "Detected two calls to `torchdynamo.optimize(...)` with a different backend compiler arguments."
-  # PYTORCH_TEST_WITH_DYNAMO=0 PYTORCH_TEST_WITH_INDUCTOR=0 pytest test/inductor
+  python test/test_modules.py --verbose
+  pytest test/test_ops_gradients.py --verbose -k "not _complex and not test_inplace_grad_acos_cuda_float64"
```
test_torch.py ?

desertfire replied:
> test_torch.py ?

Passing rate not ideal, so I will do it later, https://github.com/pytorch/torchdynamo/issues/1730

eellison commented on test/test_modules.py:
```diff
@@ -326,6 +327,7 @@ def inner_zero_grad(obj):
 
     @skipIfMps
     @modules(module_db)
+    @skipIfTorchInductor("to be fixed")
```
are you working on these ? if not we should file issues

desertfire replied:
I plan to file an umbrella issue for the top frequent failures. But I need to land this rng workaround to make it reproducible for people.

eellison commented on test/test_ops.py:
```diff
@@ -209,6 +210,7 @@ def to_cpu(arg):
     @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
     @onlyNativeDeviceTypes
     @ops(python_ref_db)
+    @skipIfTorchInductor("Takes too long for inductor")
```
how do these tests get run ? I thought we were only invoking `test/test_modules.py` and `test/test_ops_gradients.py` ?

desertfire replied:
You got me. I ran this one locally, and just decide to smuggle it here to avoid having local changes.

eellison commented on torch/_dynamo/variables/torch.py:
```diff
@@ -320,6 +320,9 @@ def get_state_from_generator():
 
             if config.fake_tensor_propagation:
+                unimplemented(
+                    "TODO: make torch.random.set_rng_state work with fake tensor"
```
NIt: `with fake tensor in aot autograd`. That's where it fails right ?

desertfire replied:
Yes, you are right. Let me update the comment.

eellison reviewed (approved):
Cool !

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add several inductor-specific skips;
3) Enable several core tests for inductor CI;

cc jansel mlazos soumith voznesenskym yanboliang penguinwu anijain2305

[ghstack-poisoned]

desertfire committed:
Update on "Enable some PyTorch core tests with inductor"


Summary:
1) Graph break on torch.random.set_rng_state since it blocks running
inductor core tests;
2) Add several inductor-specific skips;
3) Enable several core tests for inductor CI;

cc jansel mlazos soumith voznesenskym yanboliang penguinwu anijain2305

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
